oak_containers_launcher/src/qemu.rs

```diff
@@ -0,0 +1,162 @@
+//
+// Copyright 2023 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use crate::path_exists;
+use anyhow::Result;
+use clap::Parser;
+use command_fds::tokio::CommandFdAsyncExt;
+use std::{
+    io::{BufRead, BufReader},
+    os::{fd::AsRawFd, unix::net::UnixStream},
+    path::PathBuf,
+    process::Stdio,
+};
+
+/// Represents parameters used for launching VM instances.
+#[derive(Parser, Clone, Debug, PartialEq)]
+pub struct Params {
+    /// Path to the VMM binary to execute.
+    #[arg(long, value_parser = path_exists)]
+    pub vmm_binary: PathBuf,
+
+    /// Path to the stage0 image to use.
+    #[arg(long, value_parser = path_exists)]
+    pub stage0_binary: PathBuf,
+
+    /// Path to the Linux kernel file to use.
+    #[arg(long, value_parser = path_exists)]
+    pub kernel: PathBuf,
+
+    /// Path to the initrd image to use.
+    #[arg(long, value_parser = path_exists)]
+    pub initrd: PathBuf,
+
+    /// How much memory to give to the enclave binary, e.g., 256M (M stands for Megabyte, G for
+    /// Gigabyte).
+    #[arg(long)]
+    pub memory_size: Option<String>,
```

Comment:
Maybe `guest_memory_size`? Also: Can we maybe even compute this and just ask for `application_memory_size`?